src/docs/asciidoc/reactiveProgramming.adoc

```diff
@@ -306,108 +298,103 @@ Since we are in a test, we block, waiting for the processing to finish instead,
+maintainability of code. As the layers of asynchronous processes increase in both number
+and complexity, being able to compose and read code becomes increasingly difficult. As we
+saw, the callback model is simple, but one of its main drawbacks is that, for complex
+processes, you need to have a callback executed from a callback, itself nested inside
+another callback, and so on. That mess is known as *Callback Hell*. As you can guess (or
+know from experience), such code is pretty hard to go back to and reason about.
+
+Reactor offers rich composition options, wherein code mirrors the organization of the
+abstract process, and everything is generally kept at the same level (nesting is
+minimized).
+
+=== The Assembly Line Analogy
+You can think of data processed by a reactive application as moving through an assembly
+line. Reactor is both the conveyor belt and the workstations. The raw material pours from
+a source (the original `Publisher`) and ends up as a finished product ready to be pushed
+to the consumer (or `Subscriber`).
+
+The raw material can go through various transformations and other intermediary steps or
+be part of a larger assembly line that aggregates intermediate pieces together. If there
+is a glitch or clogging at one point (perhaps boxing the products takes a
+disproportionately long time), the afflicted workstation can signal upstream to limit the
+flow of raw material.
 
 === Operators
-In Reactor, operators are the workstations in our assembly analogy. Each
-operator adds behavior to a `Publisher` and wraps the previous step's
-`Publisher` into a new instance. The whole chain is thus linked, such that data
-originates from the first `Publisher` and moves down the chain, transformed by
-each link. Eventually, a `Subscriber` finishes the process. Remember that
-nothing happens until a `Subscriber` subscribes to a `Publisher`, as we see
-shortly.
-
-TIP: Understanding this can help you avoid a common mistake that would lead you
-to believe that an operator you used in your chain is not being applied. See
-this <<faq.chain,item>> in the FAQ.
-
-While the Reactive Streams specification doesn't specify operators at all, one
-of the best added values of reactive libraries like Reactor is the rich
-vocabulary of operators that they provide. These cover a lot of ground, from
-simple transformation and filtering to complex orchestration and error handling.
+In Reactor, operators are the workstations in our assembly analogy. Each operator adds
+behavior to a `Publisher` and wraps the previous step's `Publisher` into a new instance.
+The whole chain is thus linked, such that data originates from the first `Publisher` and
+moves down the chain, transformed by each link. Eventually, a `Subscriber` finishes the
+process. Remember that nothing happens until a `Subscriber` subscribes to a `Publisher`,
+as we see shortly.
+
+TIP: Understanding operators can help you avoid a common mistake that would lead you to
```

Comment:
Replace `Understanding operators` with `Understanding that operators create new instances`